### PR TITLE
Run integration tests on SauceLabs with beta browser versions, and make these beta versions non-failing

### PR DESCRIPTION
**Pull request by danielrozenberg**
This PR replaces and enables dev browser versions with their closest beta version on SauceLabs, and runs them in a separate batch that will not fail the entire Travis run. Eventually, coupled with #14835, we can send a daily summary of tests that are failing on beta browsers to the build cop, for triaging.

As an unrelated refactor, this PR also changes the `log` statements to use the backtick (`) operator to concatenate strings, instead of pluses. For your convenience I split this into 2 commits (f4aa87a is the actual change, 2922e81 is the backtick refactor)

### CODE REVIEW

File: build-system/tasks/runtime-test.js [hunk 1]
```diff
@@ -572,26 +572,68 @@ async function runTests() {
...
+        return allBatchesExitCodes;
+      }
+    }
+
+    if (browsers.dev.length) {
+      const allBatchesExitCodes = await runTestInBatchesWithBrowsers(
+          'dev', browsers.dev);
+      if (allBatchesExitCodes) {
+        log(yellow('Some tests have failed on'), cyan('dev'),
+            yellow('browsers. However, this is not a fatal error.'));
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply build-system/tasks/runtime-test.js]: Done

File: build-system/tasks/runtime-test.js [hunk 2]
```diff
@@ -80,15 +81,14 @@ function getConfig() {
...
-        // 'SL_Chrome_Dev',
-        // 'SL_Firefox_Dev',
         'SL_Safari_12',
         // TODO(amp-infra): Evaluate and add more platforms here.
         //'SL_Chrome_Android_7',
         //'SL_iOS_11',
         //'SL_iOS_12',
         //'SL_IE_11',
+        'SL_Chrome_Dev',
+        'SL_Firefox_Dev',
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply build-system/tasks/runtime-test.js]: Done

File: build-system/tasks/runtime-test.js [hunk 3]
```diff
@@ -658,24 +657,24 @@ async function runTests() {
           } else if (argv.integration) {
             flags = ' --flags=integration_tests';
           }
-          log(green('INFO: ') + 'Uploading code coverage report to ' +
-              cyan('https://codecov.io/gh/ampproject/amphtml') + ' by running ' +
+          log(green('INFO: '), 'Uploading code coverage report to',
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply build-system/tasks/runtime-test.js]: Done

File: build-system/tasks/runtime-test.js [hunk 4]
```diff
@@ -240,7 +240,7 @@ function getImports(jsFile) {
   const rootDir = path.dirname(path.dirname(__dirname));
   const jsFileDir = path.dirname(jsFile);
   imports.forEach(function(file) {
-    const fullPath = path.resolve(jsFileDir, file) + '.js';
+    const fullPath = path.resolve(jsFileDir, `${file}.js`);
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply build-system/tasks/runtime-test.js]: The previous lines break the path (`path.dirname`) so this looks fine to me

[the remark this answers is not in the document]
Comment (rsimha) [reply build-system/tasks/runtime-test.js]: SGTM if you've tested this locally to make sure the imports are correctly computed.

File: build-system/tasks/runtime-test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rsimha) [reply build-system/tasks/runtime-test.js]: Update PR title and description to reflect that we're now using beta versions.

[the remark this answers is not in the document]
Comment (danielrozenberg) [reply build-system/tasks/runtime-test.js]: Done

**Pull request merged**